include/hal/i2c_controller.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+namespace edge::aidan
+{
+
+class I2CController 
+{
+public:
+    ~I2CController();
```

Comment:
I didn't see this defined. Just default it

Reply:
nice catch, changed